### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by hechlerp**
Cleaned up a few comments here and there. Added Calendar and DatePanel components, though they're not yet set to render anywhere. Also edited TodoUtil and the rest of the Todo flux cycle to include the dated todos. 
The API request makes a GET request to the todos index route, which, as far as I can tell, is otherwise unused. It returns the current's user's todos where the date column is within given parameters, passed up in the request. 
Currently, the query won't work, as it understands the date column to be integers in milliseconds. In addition, there are other considerations before being able to add todos to a day, such as which list they are forced to go on (if they have to go on a list) and how many a day can "hold." I imagine we can simply constrain each day to 10, and simply not render the form at the bottom of the todos list, as we did with lists.
Finally, we will need to render the calendar somewhere, for which I recommend that we create a container component for both ListIndex and Calendar.

### CODE REVIEW

**Comment by lilyxl8** [comment]
@hechlerp Can you format your comments in a bulleted list?

**Comment by lilyxl8** [comment]
@hechlerp Hey, thanks for this! I opened a PR to refactor the views, let's do that first before adding the Calendar. To make sure our foundation is solid before recycling the code for Cal.
Also would you mind putting your notes in a bulleted list next time ;)

**Comment by hechlerp** [comment]
Yeah, of course. I'm going to refactor this real quick so that it matches our newest practices.

File: frontend/actions/todo_actions.js [hunk 1]
```diff
@@ -16,6 +16,14 @@ const TodoActions = {
     });
   },
 
+  todosReceivedCalendar (todos, day) {
+    AppDispatcher.dispatch({
+      actionType: TodoConstants.TODOS_RECEIVED_CALENDAR,
```
Comment (lilyxl8) [inline frontend/actions/todo_actions.js]: I think we agreed to make TodoActions & TodoStore into CalActions & CalStore? Then you can keep the name todosReceived, ListActions already has this function.

Comment (hechlerp) [inline frontend/actions/todo_actions.js]: The todos_received function does something different than we want for the calendar though. Currently in the carousel, todos are stored by hashes of relevant days, such that a given day can access its todos in constant time. I'll change the names for sure, though.

File: frontend/utils/todo_util.js [hunk 2]
```diff
@@ -22,6 +22,18 @@ const TodoUtil = {
         TodoActions.todosReceived(list.todos);
       }
     });
+  },
+
+  fetchTodos: (day) => {
```
Comment (lilyxl8) [inline frontend/utils/todo_util.js]: fetchTodosByDate: (date)

File: app/controllers/api/todos_controller.rb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 class Api::TodosController < ApplicationController
   def index
-    @todos = Todo.all
+    # For this to work, the date column needs to be an integer
```
Comment (lilyxl8) [inline app/controllers/api/todos_controller.rb]: Why did you choose this format over Ruby's [Date](https://ruby-doc.org/stdlib-2.3.1/libdoc/date/rdoc/Date.html) class?

Comment (hechlerp) [inline app/controllers/api/todos_controller.rb]: I believe storing these as integers is not only easier to work with but requires less memory. I don't think there's a reason we would need them stored as date objects in the backend, and I don't know if SQL has a equally uncomplicated Date search method for comparing one to another as it does for integers.

**Pull request closed without merging**